### PR TITLE
feat: User and organisations can Set currency

### PR DESCRIPTION
**Pull request by joshikeno**
Resolve [#2730](https://github.com/opencollective/opencollective/issues/2730)

# Description
- supported currency constants
- add currency field to EditCollectiveForm
- update langs include new currency field intl

# Screenshots
Screenshot 2020-02-28 at 19 02 54

### CODE REVIEW

**Comment by znarf** [comment]
It's WIP so I supposed it's normal that some functionality is missin: Only User and Organization should see it, disabled if Host Account.

**Comment by joshikeno** [comment]
@znarf I want to clarify, only `new` Users or Organisations can `set currency`?

**Comment by znarf** [comment]
@joshikeno Any User and Organization that are not hosts (`isHostAccount=true`), new or existing.

File: components/edit-collective/EditCollectiveForm.js [hunk 1]
```diff
@@ -483,6 +493,21 @@ class EditCollectiveForm extends React.Component {
...
+        {
+          name: 'currency',
+          type: 'select',
+          defaultValue: get(this.state.collective, 'currency'),
+          options: currencyOptions,
+          when: () => {
+            if (get(this.state.collective, 'isHost')) {
+              return false;
+            }
+            if (get(this.state.collective, 'type') === CollectiveType.COLLECTIVE) {
```
Comment (znarf) [inline components/edit-collective/EditCollectiveForm.js]: No, let's test if it's `User` or `Organization`.

Comment (joshikeno) [reply components/edit-collective/EditCollectiveForm.js]: added check `collectiveType === User || Organisation` pass

File: components/edit-collective/EditCollectiveForm.js [hunk 2]
```diff
@@ -483,6 +493,22 @@ class EditCollectiveForm extends React.Component {
...
+          name: 'currency',
+          type: 'select',
+          defaultValue: get(this.state.collective, 'currency'),
+          options: currencyOptions,
+          when: () => {
+            const _type = get(this.state.collective, 'type');
+            if (get(this.state.collective, 'isHost')) {
+              return false;
+            }
+            if (_type === CollectiveType.ORGANIZATION || _type === CollectiveType.USER) {
```
Comment (znarf) [inline components/edit-collective/EditCollectiveForm.js]: Maybe:

```[CollectiveType.ORGANIZATION, CollectiveType.USER].includes(get(this.state.collective, 'type'))```

**Comment by znarf** [comment]
Missing is the API handling. We should make sure to let users only allow updating currency in these particular cases (User and Organization that are not hosts). In the `editCollective` mutation, look at how we're doing `updateHostFee` and do something similar.

**Pull request merged**